yarn-project/simulator/src/public/executor.ts

```diff
@@ -169,4 +172,82 @@ export class PublicExecutor {
     const newWorldState = context.persistableState.flush();
     return temporaryConvertAvmResults(execution, newWorldState, result);
   }
+
+  /**
+   * These functions are currently housed in the temporary executor as it relies on access to
+   * oracles like the contractsDB and this is the least intrusive way to achieve this.
+   * When we remove this executor and have an interface that is compatible with the kernel circuits,
+   * this will be moved to sequencer-client/prover.
+   */
+
+  /**
+   * Generates a proof for an associated avm execution. This is currently only used for testing purposes,
+   * as proof generation is not fully complete in the AVM yet.
+   * @param execution - The execution to run.
+   * @returns An AVM proof and the verification key.
+   */
+  public async getAvmProof(avmExecution: PublicExecution): Promise<Buffer[]> {
+    // The paths for the barretenberg binary and the write path are hardcoded for now.
+    // We additionally need the path to a valid crs for proof generation.
+    const bbPath = '../../barretenberg/cpp';
+    const writePath = '/tmp';
+    const { args, functionData, contractAddress } = avmExecution;
+    const bytecode = await this.contractsDb.getBytecode(contractAddress, functionData.selector);
+    await Promise.all([
+      fs.writeFile(
+        `${writePath}/calldata.bin`,
+        args.map(c => c.toBuffer()),
+      ),
+      fs.writeFile(`${writePath}/avm_bytecode.bin`, bytecode!),
+    ]);
+
+    const bbBinary = spawn(`${bbPath}/build/bin/bb`, [
+      'avm_prove',
+      '-b',
+      `${writePath}/avm_bytecode.bin`,
+      '-d',
+      `${writePath}/calldata.bin`,
+      '-c',
+      `${bbPath}/srs_db/ignition`,
+      '-o',
+      `${writePath}/proof`,
+    ]);
```

Comment:
It's not clear to me what happens if any of the above steps fail. Could you catch them and throw w/something like "could not execute xxxx, path (or pwd), binary path, error code". And a test for that ideally.